protocol/binary/writer.go

```diff
@@ -201,32 +168,32 @@ func (bw *Writer) WriteValue(v wire.Value) error {
 	switch v.Type() {
 	case wire.TBool:
 		if v.GetBool() {
-			return bw.writeByte(1)
+			return bw.sw.writeByte(1)
 		}
-		return bw.writeByte(0)
+		return bw.sw.writeByte(0)
 
 	case wire.TI8:
-		return bw.writeByte(byte(v.GetI8()))
+		return bw.sw.writeByte(byte(v.GetI8()))
```

Comment:
This should use bw.sw.WriteInt8

```suggestion
		return bw.sw.WriteInt8(v.GetI8())
```